### PR TITLE
Added code to prevent grabber from opening when unsafe

### PR DESCRIPTION
**Pull request by tails618**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 1]
```diff
@@ -74,6 +74,7 @@ public enum LiftPositionStartRelative {
   private final int SMART_MOTION_SLOT = 0;
   private LiftPosition latestPosition = LiftPosition.STARTING;
   private LiftPosition desiredPosition = LiftPosition.STARTING;
+  private boolean grabberOpen = false;
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Lift.java]: add `desired` somewhere in there?

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 2]
```diff
@@ -226,11 +227,11 @@ private void goToPosition(LiftPosition pos) {
   }
 
   public void grab() {
-    grabber.set(DoubleSolenoid.Value.kForward);
+    grabberOpen = true;
   }
 
   public void drop() {
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Lift.java]: Can we rename these openGrabber closeGrabber or something?

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 3]
```diff
@@ -379,6 +380,15 @@ public void periodic() {
     } else {
       goToPosition(desiredPosition);
     }
+
+    // If the grabber is set to open and it is safe to open, open the grabber. Otherwise, close it.
+    if ((armEncoder.getPosition() > Cal.Lift.GRABBER_CLOSED_ZONE_TOP_DEGREES
+            || armEncoder.getPosition() < Cal.Lift.GRABBER_CLOSED_ZONE_BOTTOM_DEGREES)
+        && grabberOpen) {
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Lift.java]: Super minor but swap the order of grabberOpen and the other part? Marginally more readable that way IMO+

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -174,8 +174,8 @@ private void configureBindings() {
     operatorController
         .rightBumper()
         .onTrue(new InstantCommand(() -> lights.toggleCode(Lights.LightCode.CONE), lights));
-    operatorController.leftTrigger().onTrue(new InstantCommand(lift::grab, lift));
-    operatorController.leftTrigger().onFalse(new InstantCommand(lift::drop, lift));
+    operatorController.leftTrigger().onTrue(new InstantCommand(lift::closeGrabber, lift));
+    operatorController.leftTrigger().onFalse(new InstantCommand(lift::openGrabber, lift));
```
Comment (balghane) [inline src/main/java/frc/robot/RobotContainer.java]: Hm flip these I think, should default to closed

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 5]
```diff
@@ -74,7 +74,7 @@ public enum LiftPositionStartRelative {
   private final int SMART_MOTION_SLOT = 0;
   private LiftPosition latestPosition = LiftPosition.STARTING;
   private LiftPosition desiredPosition = LiftPosition.STARTING;
-  private boolean grabberOpen = false;
+  private boolean desiredGrabberClosed = false;
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Lift.java]: Default this to true

File: src/main/java/frc/robot/subsystems/Lift.java [hunk 6]
```diff
@@ -382,9 +382,9 @@ public void periodic() {
...
     // If the grabber is set to open and it is safe to open, open the grabber. Otherwise, close it.
-    if ((armEncoder.getPosition() > Cal.Lift.GRABBER_CLOSED_ZONE_TOP_DEGREES
-            || armEncoder.getPosition() < Cal.Lift.GRABBER_CLOSED_ZONE_BOTTOM_DEGREES)
-        && grabberOpen) {
+    if (desiredGrabberClosed
+        && (armEncoder.getPosition() > Cal.Lift.GRABBER_CLOSED_ZONE_TOP_DEGREES
+            || armEncoder.getPosition() < Cal.Lift.GRABBER_CLOSED_ZONE_BOTTOM_DEGREES)) {
       grabber.set(Value.kForward);
     } else {
       grabber.set(Value.kReverse);
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/Lift.java]: This logic isn't right anymore. We should be closed if desired OR if opening is unsafe

**Review by balghane — CHANGES_REQUESTED** [review]

**Review by avakimc — APPROVED** [review]
I am Ava and I approve this PR 👍

**Comment by Panda8188** [comment]
I am Sid and I approve the approval of this PR by Ava 👍🏾

File: src/main/java/frc/robot/subsystems/Lift.java (hunk 6 above)
Comment (tails618) [reply src/main/java/frc/robot/subsystems/Lift.java]: I've made it so if the variable is false (we want it to be open) and its safe it sets to open.

**Review by balghane — APPROVED** [review]

**Pull request merged**